src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 200) {
+    this.health = health;
+    this.name = 'Animal';
+  }
+
+  dead(healthPoints) {
+    this.health += healthPoints;
+
+    if (this.health < 100) {
+      Animal.alive = Animal.alive.filter((animal) => animal.health);
+    }
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, 100) {
+    this.hidden = 'hidden';
+  }
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(mammal) {
+    if (
+      !mammal.hidden
+    ) {
+      mammal.dead(50 * 0);
```

Comment:
[CODE PERFORMANCE] Avoid using nested loops for removing dead animals from the alive array. The current implementation does not have nested loops, but the logic for removing dead animals is incorrect and should be revised.